lucene/core/src/java/org/apache/lucene/search/KnnExactDeletionCondition.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+/**
+ * A query for deleting the exact values of the specified querying vector, used for
+ * {@link org.apache.lucene.index.IndexWriter#deleteDocuments(Query...)}.
+ */
+public class KnnExactDeletionCondition extends KnnGraphQuery {
```

Comment:
Why do we need to include the word "deletion" in the Query name? If my understanding is correct this finds a set of documents that have exact values to the query vector, and does not concern what the user do with the query results. Additionally it seems slightly odd to me that the (public) Query subclass does not obey the naming convention (`XXXQuery`). I would prefer the name "KnnExactVectorQuery" or something like that here...

If you intend to use this Query class for special purpose ("for deleting documents"), I'd suggest you don't expose it as a public class but firstly keep it as a dedicated inner class within another class. Would it be possible?

Reply:
Yes, your understanding is correct and highly appreciated. 
1. The class KnnExactVectorValueQuery (revised name) is originally designed for deleting vectors by query. When we are going to delete some vector by its exact values, Lucene seems to retrieve all the similar vectors and then delete all of them, which may not be the behavior we expect. The class KnnExactVectorValueWeight is the corresponding weight for KnnExactVectorValueQuery to filter the vectors that not exactly equal to the query vector. Therefore, the two classes are designed for dedicated usage.
2. I'm not pretty sure if there is any situation that we need to delete by queries. So I moved both classes into my test case.
Now, there isn't any new feature, just some minor bug fix.